### PR TITLE
chore: archive @opentelemetry/browser-extension-autoinjection

### LINKED ISSUE #1101: resource-detector-github failed to compile
Constant CI failure with the following outputs:

```
> @opentelemetry/resource-detector-github@0.27.0 compile
> tsc -p .

error TS2688: Cannot find type definition file for 'parse5'.
  The file is in the program because:
    Entry point for implicit type library 'parse5'

../../../node_modules/@types/jsdom/base.d.ts:4:33 - error TS7016: Could not find a declaration file for module 'parse5'. '/home/runner/work/opentelemetry-js-contrib/opentelemetry-js-contrib/node_modules/parse5/lib/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/parse5` if it exists or add a new declaration (.d.ts) file containing `declare module 'parse5';`

4 import { ElementLocation } from "parse5";
                                  ~~~~~~~~
```

Example link: https://github.com/open-telemetry/opentelemetry-js-contrib/runs/7547491100?check_suite_focus=true
Find out more: https://github.com/open-telemetry/opentelemetry-js-contrib/actions/workflows/unit-test.yml

### PR DESCRIPTION
**Pull request by dyladan**
Fixes #1101

Related #1080 

Moved to archive instead of simply deleting. Git does make it easy to bring packages back, but they are not easily discoverable if they're removed entirely from main.

### CODE REVIEW

File: archive/opentelemetry-browser-extension-autoinjection/package.json
Comment (Flarna) [inline archive/opentelemetry-browser-extension-autoinjection/package.json:60]: Is this change intended?

**Review by legendecas — APPROVED** [review]
Let's do this to unblock other PRs.

**Review by Flarna — APPROVED** [review]

**Review by vmarchaud — APPROVED** [review]

**Comment by vmarchaud** [comment]
ping @svrnm since you wrote it i believe

File: archive/opentelemetry-browser-extension-autoinjection/package.json
Comment (dyladan) [reply archive/opentelemetry-browser-extension-autoinjection/package.json:60]: I was trying to fix the build before archiving. Turned out not to be easy. I'll revert this.

**Comment by svrnm** [comment]
@vmarchaud: apologies! I missed that one since I was out of office around the occurrence of that issue with the browser extension and got flooded with other issues on github. 

Yes I wrote it, but didn't find the time over the last few months to get back to it & keep it maintained.

I am happy to carve out some hours to clean things up, if there's still interest to have that extension within the js-contrib repo.

**Comment by dyladan** [comment]
We would love to have it but it was causing build problems which blocked other prs. At the time we didn't have a better course of action. If you can bring it back to life then I would be happy to have it back if the other maintainers agree. We haven't really come up with an official policy yet but we will need to as these issues may be more common in the future.

**Comment by svrnm** [comment]
Thanks, if there are no major objections from any other maintainer I will see to get it back working.

**Comment by legendecas** [comment]
I support bringing this back :D thank you for working on this!

**Pull request merged**
